airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -112,22 +117,170 @@ def test_connection(self) -> tuple[bool, str]:
         """
         client = self.get_client()
         client.schema.create(schema_json)
 
+    @retry(reraise=True, stop=stop_after_attempt(3), retry=retry_if_exception_type(requests.ConnectionError))
+    def get_schema(self, class_name: str | None = None):
+        """Get the schema from Weaviate.
+
+        get schema of a class or all classes.
+        """
+        client = self.get_client()
+        return client.schema.get(class_name)
+
+    @retry(reraise=True, stop=stop_after_attempt(3), retry=retry_if_exception_type(requests.ConnectionError))
+    def delete_class(self, class_name: str):
+        """Delete a schema class from Weaviate. This deletes all associated data."""
+        client = self.get_client()
+        client.schema.delete_class(class_name)
+
+    def delete_schema(
+        self, class_names: list[str] | str | None = None
+    ) -> list[UnexpectedStatusCodeException] | None:
+        """
+        Deletes all or specific class if class_names are provided.
+
+        If no class_name is given remove the entire schema from the Weaviate instance and all data associated
+        with it. If class_names are given, delete schema classes from Weaviate. This deletes all associated data.
+        :return: list of error object if any
+        """
+        client = self.get_client()
+        class_names = [class_names] if class_names and isinstance(class_names, str) else class_names
+        if class_names:
+            error_list = []
+            for class_name in class_names:
+                try:
+                    self.delete_class(class_name=class_name)
+                except UnexpectedStatusCodeException as e:
+                    error_list.append(e)
+            return error_list
+        else:
+            return client.schema.delete_all()
+
+    @retry(reraise=True, stop=stop_after_attempt(3), retry=retry_if_exception_type(requests.ConnectionError))
+    def update_class(self, class_name: str, config: dict):
+        """Update schema's class."""
+        client = self.get_client()
+        client.schema.update_config(class_name=class_name, config=config)
+
+    def update_multiple_classes(self, schema_json: list[dict]) -> list[UnexpectedStatusCodeException] | None:
+        """Updated multiple classes.
+
+        :param schema_json: list of class_config objects
+            .. seealso:: `example of class_config <https://weaviate-python-client.readthedocs.io/en/v3.25.2/weaviate.schema.html#weaviate.schema.Schema.update_config>`_.
+        :return: list of error object if any
+        """
+        error_list = []
+        for config in schema_json:
+            try:
+                self.update_class(class_name=config.pop("class"), config=config)
+            except UnexpectedStatusCodeException as e:
+                error_list.append(e)
+        return error_list
+
+    def upsert_classes(self, schema_json: dict[str, Any] | str, existing: ExitingSchemaOptions = "ignore"):
```

Comment:
There seems to be a lot of redundancy for creation of classes. We have create_schema, create_class, update_class, update_multiple_classes, upsert_classes all trying to manipulate classes 

We should try to consolidate them instead of building separate methods.

Reply:
I think we don't need an upsert method in a hook, a task/operator should do an upsert, not a hooks method. We can provide various methods that would help a user to write an upsert task/operator.

Reply:
I don't think all the methods listed are just creating classes, but I do see some redundancy. 

1. Well `create_schema` is not the same thing as `create_class` because one is super set and the other is a subset. But I guess the user can use `create_schema` to create classes as well. So maybe we can get rid of `create_class`. But the usage of that would look weird when creating a single class with the `create_schema` method. WDYT? 

        For example: 
        ```
        create_schema({"classes": <class dict>})
        ```

2. For `update_class` and `update_multiple_classes`, we can get rid of `update_multiple_classes`. 

3. I think `upsert_classes` should stay, as it handles a different scenario.

Reply:
I think create class, update_class and upsert_classes can all be combined within upsert_classes.

Upsert = create + update, right?

Reply:
Since `create_class` exposes `client.schema.create_class(class_json)` and `update_class` exposes  `client.schema.update_config` we are not using either of them in the `upsert_classes` methods. I'm afraid its gonna be difficult to merge these.

Also, I checked the possibility of using `create_class` and `update_class` in `upsert_classes`. But there are issues:

1. `create_class` doesn't allow us to create multiple classes at once, hence we are using `create_schema`
2. `update_class` (now changed to `update_config`) doesn't allow us to update class properties. So we are deleting and recreating classes in `upsert_classes` methods.